src/test/groovy/org/osiam/auth/login/internal/InternalAuthenticationProviderSpec.groovy

```diff
@@ -23,15 +23,96 @@
  */
 package org.osiam.auth.login.internal
 
+import org.osiam.resources.provisioning.SCIMUserProvisioning
+import org.osiam.resources.scim.User
+import org.springframework.security.authentication.BadCredentialsException
+import org.springframework.security.authentication.encoding.ShaPasswordEncoder
+import org.springframework.security.core.Authentication
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
+import spock.lang.Shared
 import spock.lang.Specification
 
 class InternalAuthenticationProviderSpec extends Specification {
 
-    def 'the internal provider only supports InternalAuthentication class'() {
-        given:
-        InternalAuthenticationProvider provider = new InternalAuthenticationProvider(null, null, null, null)
+    // hashs of 'password'
+    def VALID_BCRYPT_HASHED_PASSWORD = '$2a$13$PfXyvWHogpPZgZjnk2a0eO7wSSNsMZ6DL4x5k8Bk0CK1cwXwMs0de'
+    def VALID_SHA_HASHED_PASSWORD = 'dc9b4c4a06e7dc774e469a16243ff448a02932fd323bb7ecb0ece308d6bb13c964dd7a0b7b99b15dc785004bdb869450f4070a9f83b027af43025149d1bc8f34'
+
+    // hash of 'wrongpassword'
+    def WRONG_SHA_HASHED_PASSWORD = 'bfe2ee0a1bd6ca719f7b6855eaa2ce166e9f06678a662accac9b8f84ba84801d7f76c7c9a05d5207301fee9a6a11d3178eb34c47647835e7f888d3ce6e7935ba'
+
+    def userName = 'userName'
+    def password = 'password'
+    def userId = '618b398c-0110-43f2-95df-d1bc4e7d2b4a'
+
+    SCIMUserProvisioning userProvisioning = Mock()
```

Comment:
can you please use `def userprovisioning = Mock(SCIMUserProvisioning)`? Intellij is capable to infer the type and it's more 'Groovy'